core/src/autogluon/core/searcher/bayesopt/autogluon/constrained_gp_fifo_searcher.py

```diff
@@ -0,0 +1,262 @@
+from typing import Callable, Dict, Type, Optional, Any
```

Comment:
A general thought: This is a lot of changes, do we think it makes sense to have an entire submodule `autogluon.search` dedicated to this code / searcher code in general? This could be an optional dependency for other modules which would allow to plug-in searcher backends. This would lighten the `autogluon.core` code and ensure that the blast radius of searcher changes are contained within the searcher module.

Reply:
This would also make migration easier if we eventually move the searcher logic to ray or some similar package.

Reply:
In fact, we will move all this code out of the AutoGluon codebase and bring it back as a dependence. We are currently working on this full steam, our code would then work on AutoGluon, but also on Ray Tune. In the end, this is the cleanest solution.

This will happen likely in April, latest in May.

Of course, we'll make sure everything will work just as before.

Reply:
That's great, I think that is the cleanest approach as well.